testkit/jvm/src/main/scala/zio/testkit/TestClock.scala

```diff
@@ -38,13 +44,26 @@ case class TestClock(ref: Ref[TestClock.Data]) extends Clock.Service[Any] {
 
   final def adjust(duration: Duration): UIO[Unit] =
     ref.update { data =>
-      Data(data.nanoTime + duration.toNanos, data.currentTimeMillis + duration.toMillis, data.sleeps0)
+      Data(
+        data.nanoTime + duration.toNanos,
+        data.currentTimeMillis + duration.toMillis,
+        data.sleeps0,
+        TestClock.offset(data.currentDateTime.toInstant.toEpochMilli + duration.toMillis)
+      )
     }.unit
 
 }
 
 object TestClock {
-  val Zero = Data(0, 0, Nil)
+  val Zero = Data(0, 0, Nil, offset(0))
+
+  def offset(millis: Long): OffsetDateTime =
+    OffsetDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneId.of("Z"))
 
-  case class Data(nanoTime: Long, currentTimeMillis: Long, sleeps0: List[Duration])
+  case class Data(
+    nanoTime: Long,
+    currentTimeMillis: Long,
+    sleeps0: List[Duration],
+    currentDateTime: OffsetDateTime
```

Comment:
There is duplicated information now across `currentTimeMillis` and `currentDateTime`. Instead, just store a `timezone`, you can default to UTC, and construct `currentDateTime` from `currentTimeMillis` and from `timezone`.